pinot-spi/src/main/java/org/apache/pinot/spi/data/DateTimeGranularitySpec.java

```diff
@@ -87,33 +85,43 @@ public TimeUnit getTimeUnit() {
    * <ul>
    * <li>Convert a granularity to millis.
    * This method should not do validation of outputGranularity.
-   * The validation should be handled by caller using {@link #isValidGranularity(String)}</li>
+   * The validation should be handled by caller using {@link #isValidGranularity}</li>
    * <ul>
    * <li>1) granularityToMillis(1:HOURS) = 3600000 (60*60*1000)</li>
    * <li>2) granularityToMillis(1:MILLISECONDS) = 1</li>
    * <li>3) granularityToMillis(15:MINUTES) = 900000 (15*60*1000)</li>
    * </ul>
    * </ul>
-   * @return
    */
   public Long granularityToMillis() {
     return TimeUnit.MILLISECONDS.convert(_size, _timeUnit);
   }
 
   /**
    * Check correctness of granularity of {@link DateTimeFieldSpec}
-   * @param granularity
-   * @return
    */
-  public static boolean isValidGranularity(String granularity) {
-    Preconditions.checkNotNull(granularity);
+  public static boolean isValidGranularity(String granularity, Logger ctxLogger) {
```

Comment:
Recommend using exception to pass the failure message instead of passing in a logger. The problem of using logger is that the caller cannot control the logging level. For user input, in most cases we don't want to log error which usually stands for severe problems.
```suggestion
  public static void validateGranularity(String granularity) {
```